service/history/fx.go

```diff
@@ -170,6 +173,31 @@ func TelemetryInterceptorProvider(
 	)
 }
 
+func HealthSignalAggregatorProvider(
+	dynamicCollection *dynamicconfig.Collection,
+	metricsHandler metrics.Handler,
+	logger log.ThrottledLogger,
+) interceptor.HealthSignalAggregator {
+	return interceptor.NewHealthSignalAggregatorImpl(
+		dynamicconfig.PersistenceHealthSignalWindowSize.Get(dynamicCollection)(),
+		dynamicconfig.PersistenceHealthSignalBufferSize.Get(dynamicCollection)(),
+		logger,
+	)
+}
+
+func HealthCheckInterceptorProvider(
+	dynamicCollection *dynamicconfig.Collection,
+	healthSignalAggregator *interceptor.HealthSignalAggregatorImpl,
+) *interceptor.HealthCheckInterceptor {
+	if dynamicconfig.HistoryHealthSignalMetricsEnabled.Get(dynamicCollection)() {
+		return interceptor.NewHealthCheckInterceptor(
+			healthSignalAggregator,
+		)
+	}
+	return interceptor.NewHealthCheckInterceptor(
+		interceptor.NoopHealthSignalAggregator,
+	)
```

Comment:
Is it possible to check HistoryHealthSignalMetricsEnabled so it doesn't require reboot when updated the config.